src/main/java/gift/security/JwtProvider.java

```diff
@@ -0,0 +1,67 @@
+package gift.security;
+
+import gift.domain.Member;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.stereotype.Component;
+
+import java.security.Key;
+import java.util.Date;
+
+@Component
+public class JwtProvider {
+
+    private static final String SECRET_KEY = "Yn2kjibddFAWtnPJ2AFlL8WXmohJMCvigQggaEypa5E=";
+
+    private static final long EXPIRATION_TIME = 1000 * 60 * 60;
+
+    private final Key key;
+
+    public JwtProvider() {
+        this.key = Keys.hmacShaKeyFor(Decoders.BASE64.decode(SECRET_KEY));
+    }
+
+    public String generateToken(Member member) {
+        return Jwts.builder()
+                .setSubject(member.getId().toString())
+                .claim("email", member.getEmail())
+                .claim("role", member.getRole())
+                .setIssuedAt(new Date())
+                .setExpiration(new Date(System.currentTimeMillis() + EXPIRATION_TIME))
+                .signWith(key, SignatureAlgorithm.HS256)
+                .compact();
```

Comment:
`new Date()`, `System.currentTimeMillis()` 
둘 다 현재 호출 시점에 시각을 얻는 메서드라서 두 값이 달라요.

이렇게 시스템 시간을 여러 번 얻는 것이 대부분의 상황에서 큰 문제가 되진 않지만,
테스트나 정확성이 중요한 보안 로직에서 버그를 유발할 가능성이 있기 때문에, 
하나의 기준 시각을 변수로 잡아 쓰는 습관을 가지시는 것이 좋습니다!

```suggestion
        Date now = new Date();
        Date expirationDate = new Date(now.getTime() + EXPIRATION_TIME);

        return Jwts.builder()
                .setSubject(member.getId().toString())
                .claim("email", member.getEmail())
                .claim("role", member.getRole())
                .setIssuedAt(now)
                .setExpiration(expirationDate)
                .signWith(key, SignatureAlgorithm.HS256)
                .compact();
```